File: projects/04/Factorial.java
/* Factorial.java displays the factorial of a given number
 *
 * Input: number n.
 * PRE: numer must be greater than zero
 * Output: A value that is the factorial of the number input, n
 * 
 * Begun by: Nelesen, for CS 214 at Calvin College.
 * Completed by: Sinai Park(sp46)
 * Date: March 2 2020
 ********************************************************/
import java.util.Scanner; // Get input

class Factorial{

  public static void main(String[] args) {
     
    Scanner keyboard = new Scanner(System.in);
    System.out.println("\nTo compute n!, enter n: ");
    double number = keyboard.nextDouble(); //the input integer is the number being calculated
   
    // computing the factorial in a for loop

    double result = 1;                 //store resulting factorial in result
    for (double i = number; i > 0; i--)//as i is number, start from it and decrement
    {
      result *= i;    //compute factorial in the for loop
    }
      System.out.println("The factorial of " + number + " is " ); //system output
      System.out.println(result);                                 //print the factorial
      
  }
}

File: projects/04/clojure/src/factorial.clj
;;;; logTable.clj displays a table of logarithms.
;;;;
;;;; Input: a number, float.
;;;; Precondition: the number must be a positive integer.
;;;; Output: The nth factorial of the integer value n
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Sinai Park(sp46)
;;;; Date: March 4 2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns factorial)                                 ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;; Factorial() recursively generates the nth factorial of a number.
;;;; Receive: the nth count as it decrements by one each loop
;;;;
;;;; Precondition: count value must be greater than zero.
;;;;                factorial number must start at one
;;;; Output: The factorial of given number.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

;; direct-recursive version

; replace this line with the definition of Factorial().

(defn factorialCalc [number result count] 
  (if (<= count number)      
    (do 
      (factorialCalc number (* result count) (+ count 1)) ;recursion
     
  )
  result ;show factorial stores in result variable
  )
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; -main is a 'driver' for the displayLogTable() functions.
;;; Input: n, integer value.
;;; Output: The factorial of the integer value n, as user input
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn -main []
 (print "To compute the factorial of given n,enter the value: ") (flush)
  (let  
    [number (read)]
    (print(factorialCalc number 1.0 2.0))
    (println)
  )
 )
